src/pages/SignUp.js

```diff
@@ -0,0 +1,168 @@
+  const [CPF, setCPF] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [isSignUpSucess, setIsSignUpSucess] = useState(false);
+  const history = useHistory();
+  const { alertMessage, setAlertMessage } = useContext(InputContext);
+  const passwordRegex = '^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$';
+  const passwordRules = 'Your password must contain at least 8 characters, 1 upper case letter, 1 lower case letter, 1 number and 1 special character.';
+  const stringWithOnlyNumbers = '^[0-9]+$';
+
+  useEffect(() => {
+    setAlertMessage(passwordRules);
+  }, []);
+
+  const validateRepeatedPassword = () => {
+    if (password !== passwordConfirmation) {
+      setLoading(false);
+      setAlertMessage('Your password confirmation is wrong!');
+      setTimeout(() => setAlertMessage(passwordRules), 6000);
+      return false;
+    }
+    return true;
+  };
+
+  function signUpRequest(event) {
+    event.preventDefault();
+    setLoading(true);
+
+    const isRepeatedPasswordValid = validateRepeatedPassword();
+    if (!isRepeatedPasswordValid) return;
+
+    const signUpBody = {
+      name,
+      email,
+      password,
+      CPF,
+    };
+    signUp(signUpBody)
+      .then(() => {
+        setLoading(false);
+        setIsSignUpSucess(true);
+      })
+      .catch((err) => {
+        setAlertMessage(err.response?.data);
+        setTimeout(() => setAlertMessage(passwordRules), 6000);
+        setLoading(false);
+      });
+  }
+
+  return (
+    <StyledPageContainer>
+      <StyledSignContent>
+        {isSignUpSucess ? (
+          <StyledSignUpSucess>
+            <StyledSignUpSucessMessage>
+              Your account has been created! Now you just have to sign in.
+            </StyledSignUpSucessMessage>
+            <div className="timer-wrapper">
+              <CountdownCircleTimer
+                isPlaying
+                duration={5}
+                colors={[
+                  ['#ffffff'],
+                ]}
+                size={140}
+                trailColor="#5d1919"
+                strokeWidth={12}
+                onComplete={() => history.push('/sign-in')}
+              >
+                {({ remainingTime }) => remainingTime}
+              </CountdownCircleTimer>
+            </div>
+          </StyledSignUpSucess>
+        ) : (
+          <>
+            <StyledLogo>
+              <h1>BookLand</h1>
+              <h2>A fictional bookstore that sells fictional books!</h2>
+            </StyledLogo>
+            <StyledForm
+              onSubmit={(e) => {
+                setLoading(true);
+                signUpRequest(e);
+              }}
+            >
+              <StyledInput
+                placeholder="Name"
+                type="text"
+                value={name}
+                onChange={(e) => setName(e.target.value)}
+                minLength="1"
+                maxLength="30"
+                required
```

Comment:
Are the inputs still enabled after the person click on the button to submit? If it is, I think it's better if they aren't so that the person can't change the info as the page is loading.